packages/apollo/src/components/OrdererDetails/ChannelParticipationDetails.js

```diff
@@ -27,24 +27,116 @@ import Logger from '../Log/Logger';
 import SVGs from '../Svgs/Svgs';
 import ChannelParticipationModal from './ChannelParticipationModal';
 import ChannelParticipationUnjoinModal from './ChannelParticipationUnjoinModal';
+import { MspRestApi } from '../../rest/MspRestApi';
+import { NodeRestApi } from '../../rest/NodeRestApi';
+import { OrdererRestApi } from '../../rest/OrdererRestApi';
+import ChannelModal from '../ChannelModal/ChannelModal';
+import ChannelApi from '../../rest/ChannelApi';
+import NodeStatus from '../../utils/status';
 import JoinOSNChannelModal from '../JoinOSNChannelModal/JoinOSNChannelModal';
 import _ from 'lodash';
 import emptyImage from '../../assets/images/empty_channels.svg';
 import ActionsHelper from '../../utils/actionsHelper';
 
+
 const naturalSort = require('javascript-natural-sort');
 const SCOPE = 'ChannelParticipationDetails';
 const Log = new Logger(SCOPE);
+const semver = require('semver');
 
 class ChannelParticipationDetails extends Component {
+
+	constructor(props) {
+		super(props);
+		this.initialized = false;
+		this.channel = {};
+		this.orderers = [];
+		this.allOrderers = [];
+		this.allPeers = [];
+		this.msp_id = null;
+		this.acls = [];
+		this.anchorPeers = [];
+		this.chaincode = [];
+		this.nOutOf = {
+			n: 1,
+			outOf: 1,
+		};
+		this.blockParams = {
+			absolute_max_bytes: null,
+			max_message_count: null,
+			preferred_max_bytes: null,
+			timeout: null,
+		};
+		this.raftParams = {
+			tick_interval: null,
+			election_tick: null,
+			heartbeat_tick: null,
+			max_inflight_blocks: null,
+			snapshot_interval_size: null,
+		};
+		this.existingLifecyclePolicy = {
+			type: 'MAJORITY',
+			members: [],
+			n: '',
+		};
+		this.existingEndorsementPolicy = {
+			type: 'MAJORITY',
+			members: [],
+			n: '',
+		};
+		this.props.updateState(SCOPE, {
+			members: [],
+			ordererMembers: [],
+			anchorPeersLoading: false,
+			peerList: [],
+			ordererList: [],
+			joinChannelModal: false,
+			deleteConsenter: null,
+			updateConsenter: null,
+			ordererMSPs: [],
+			ordererCheck: true,
+		});
+		this.capabilities = {};
+		this.consenters = [];
+		this.ordererHost = '';
+	}
+
 	componentDidMount() {
 		this.props.updateState(SCOPE, {
 			showCPDetailsModal: false,
+			showEditChannelModal: false,
 			showCPUnjoinModal: false,
 			createChannelModal: false,
+			disableSubmit: true,
 		});
 	}
 
+	showEditChannelModal = async (channel) => {
+		this.props.updateState(SCOPE, {
+			loading: true,
+			editLoading: true,
+		});
+		let channel_details = await ChannelApi.getChannel(channel.name);
+		this.props.updateState(SCOPE, {
+			channelEdit: channel.name,
+			peerName: channel_details.peers[0].id,
```

Comment:
this won't work if we don't have peers, and this feature should still work even if we don't have any peers. we don't need a peer to load the edit channel modal, but you will need to do some work to get the config block from the alternate api.

so what i mean is ,we are currently calling `getChannelConfig() in` `ChannelApi.js`, which calls `window.stitch.getChannelConfigFromPeer()` from the stitch sdk, instead of that, you can call `window.stitch.getChannelBlockFromOrderer()`, but in this new flow. the old flow should still call the peer to get the config block.

Reply:
check out stitch's readme for details on usage: https://github.com/hyperledger-labs/fabric-operations-console/tree/main/packages/stitch/docs#getChannelConfigBlockFromOrderer